source/include/FreeRTOS_DNS.h

```diff
@@ -41,43 +41,29 @@
 #endif
 /* *INDENT-ON* */
 
-/*
- * LLMNR is very similar to DNS, so is handled by the DNS routines.
- */
-uint32_t ulDNSHandlePacket( const NetworkBufferDescriptor_t * pxNetworkBuffer );
+/* The LLMNR MAC address is 01:00:5e:00:00:fc */
+extern const MACAddress_t xLLMNR_MacAddress;
 
-#if ( ipconfigUSE_LLMNR == 1 )
-    /* The LLMNR MAC address is 01:00:5e:00:00:fc */
-    extern const MACAddress_t xLLMNR_MacAddress;
-#endif /* ipconfigUSE_LLMNR */
-
-#if ( ipconfigUSE_LLMNR == 1 ) && ( ipconfigUSE_IPv6 != 0 )
+/* The LLMNR IPv6 MAC address is 33:33:00:01:00:03 */
+extern const MACAddress_t xLLMNR_MacAddressIPv6;
 
 /* The LLMNR IPv6 address is ff02::1:3 */
-    extern const IPv6_Address_t ipLLMNR_IP_ADDR_IPv6;
-
-/* The LLMNR IPv6 MAC address is 33:33:00:01:00:03 */
-    extern const MACAddress_t xLLMNR_MacAddressIPv6;
-#endif /* ipconfigUSE_LLMNR */
+extern const IPv6_Address_t ipLLMNR_IP_ADDR_IPv6;
 
-#if ( ipconfigUSE_MDNS == 1 )
-    /* The MDNS MAC address is 01:00:5e:00:00:fc */
-    extern const MACAddress_t xMDNS_MacAddress;
-#endif /* ipconfigUSE_MDNS */
+/* The MDNS MAC address is 01:00:5e:00:00:fc */
+extern const MACAddress_t xMDNS_MacAddress;
 
-#if ( ipconfigUSE_MDNS == 1 ) && ( ipconfigUSE_IPv6 != 0 )
+/* The MDNS IPv6 MAC address is 33:33:00:01:00:03 */
+extern const MACAddress_t xMDNS_MACAddressIPv6;
```

Comment:
I feel like xMDNS_**MAC**AddressIPv6 and xMDNS_**Mac**AddressIPv6 at line 64 are used as a same variable. According to PR recently merged we wanted to keep _Mac instead of _MAC for uniformity. Is there any reason to introduce xMDNS_**MAC**AddressIPv6?
Also i see that const MACAddress_t xMDNS_MacAddressIPv6 has been removed but is still used in some of the networkInterface.c files which can cause issue. Can you please check once.

Reply:
That looks like a mistake to me.

We said that "_MacAddress" is easier to read than "_MACAddress". 

This was in my reason [PR 1169](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1169):
~~~diff
     const MACAddress_t xLLMNR_MacAddress;
     const MACAddress_t xLLMNR_MacAddressIPv6;
     const MACAddress_t xMDNS_MacAddress;
-    const MACAddress_t xMDNS_MACAddressIPv6;
+    const MACAddress_t xMDNS_MacAddressIPv6;

+    /* Guarantee backward compatibility. */
+    #define xMDNS_MACAddressIPv6    xMDNS_MacAddressIPv6
~~~
So we should not create a variable called `xMDNS_MACAddressIPv6`.

( the day I wrote PR 1169, I had lost an hour of searching for a const variable called `xMDNS_MacAddressIPv6`. Frustrating. )

Thanks

Reply:
Great catch! I've updated the error part here.